daffodil-lib/src/main/scala/org/apache/daffodil/xml/QNameBase.scala

```diff
@@ -224,8 +224,7 @@ trait QNameBase extends Serializable {
    * must show up in diagnostic messages. Mistakes by having the wrong prefix
    * or by omitting one, are very common.
    */
-  def prefix: Option[String]
-  def prefixOrNull: String = prefix.orNull
+  def lexicalPrefix: Option[String]
```

Comment:
Add scaladoc of what you mean by lexicalPrefix.

The term lexical is often used to mean "due to surrounding textual context", but I think your intent here is to be clear when we're dealing with the prefix part of a QName, from a prefix that is somehow implied by things elsewhere, like lexically enclosing xmlns default namespace declarations when there is also a prefix for that same namespace, or elementFormDefault on a schema so an element's name will have a namespace and that could have a lexically surrounding xmlns binding, etc. 

Either of these concepts can be considered "lexical", so ... scaladoc.